src/cranelift_backend/mod.rs

```diff
@@ -163,6 +163,33 @@ impl CodeGen for hir::If {
     }
 }
 
+impl CodeGen for hir::Else {
+    fn codegen<'a>(&'a self, context: &mut Context<'a>, builder: &mut FunctionBuilder) -> Value {
+        let expr = builder.create_block();
```

Comment:
Similar to the llvm code - we don't need a new block for `expr` - it can be evaluated in the current block since we don't need to conditionally jump to it or anything.

You can think of `a else b` as desugaring to:
```hs
match a
| Some value -> value
| None -> b
```
Which in cranelift is something like the following pseudocode:
```llvm
v0, v1, .., vN = ... evaluate a ...     ; cranelift doesn't have structs, so all values are always "unpacked"
                                        ; because of this v0 is the tag value, and the N values after it are the inner payload of the Maybe
brnz v0, some_block    ; if the tag is non-zero, go to some_block

some_block:
  jump end(v1, .., vN)  ; jump to end with the arguments v1 to vN (the inner Maybe values)

none_case:
  v2, .., v2N = ... evaluate b ...
  jump end(v2, .., v2N)      ; You should get N values from evaluating b as well since it's the same type as the inner maybe values above

end(v3, .., v3N):   ; results of the match are block arguments (cranelift uses these instead of Phi)
  ... block continues ...
```

You can always use `ante --emit ir --backend craneilft <filename>` to see what cranelift you're emitting for a particular file